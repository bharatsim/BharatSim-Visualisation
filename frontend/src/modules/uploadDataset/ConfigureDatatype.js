import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@material-ui/core';
import { getFileExtension } from '../../utils/fileUploadUtils';
import CsvPreview from './CsvPreview';
import JsonPreview from './jsonPreview';
import { EXTENDED_JSON_TYPES } from '../../constants/fileUpload';

function ConfigureDatatype({ selectedFile, previewData, schema }) {
  function getViewerComponent() {
    if (EXTENDED_JSON_TYPES.includes(getFileExtension(selectedFile))) {
      return <JsonPreview selectedFile={selectedFile} previewData={previewData} />;
    }
    if (getFileExtension(selectedFile) === 'csv') {
      return <CsvPreview schema={schema} previewData={previewData} selectedFile={selectedFile} />;
    }
    return `Cannot Preview Data  for ${selectedFile.name}`;
  }

  return <Box>{getViewerComponent()}</Box>;
}

ConfigureDatatype.propTypes = {
  selectedFile: PropTypes.objectOf(File).isRequired,
  previewData: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.shape({})), PropTypes.shape({})])
    .isRequired,
  schema: PropTypes.shape({}).isRequired,
};

export default ConfigureDatatype;
