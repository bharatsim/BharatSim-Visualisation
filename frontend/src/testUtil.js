// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
