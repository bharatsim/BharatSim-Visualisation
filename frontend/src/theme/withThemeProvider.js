import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
