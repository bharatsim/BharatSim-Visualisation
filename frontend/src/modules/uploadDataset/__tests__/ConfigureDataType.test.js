import React from 'react';
import ConfigureDatatype from '../ConfigureDatatype';
import withThemeProvider from '../../../theme/withThemeProvider';
import { renderWithRedux as render } from '../../../testUtil';

jest.mock('../../../uiComponent/table/Table');

const Component = withThemeProvider(ConfigureDatatype);

describe('<ConfigureDatatype />', () => {
  it('should match snapshot for csv file preview', () => {
    const { container } = render(
      <Component
        previewData={[
          { col1: 'abc', col2: 123 },
          { col1: 'abcd', col2: 1234 },
        ]}
        schema={{ col1: 'string', col2: 'number' }}
        selectedFile={new File([''], 'filename.csv')}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot for json file preview', () => {
    const { container } = render(
      <Component
        previewData={[
          { col1: 'abc', col2: 123 },
          { col1: 'abcd', col2: 1234 },
        ]}
        schema={{ col1: 'string', col2: 'number' }}
        selectedFile={new File([''], 'filename.json')}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should match snapshot for invalid file preview', () => {
    const { container } = render(
      <Component
        previewData={[
          { col1: 'abc', col2: 123 },
          { col1: 'abcd', col2: 1234 },
        ]}
        schema={{ col1: 'string', col2: 'number' }}
        selectedFile={new File([''], 'filename.error')}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
