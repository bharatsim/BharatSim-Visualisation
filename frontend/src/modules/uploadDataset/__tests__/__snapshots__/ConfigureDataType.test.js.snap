// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigureDatatype /> should match snapshot for csv file preview 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <div>
        Table
        <div>
          {
  "title": "DataFile",
  "columns": [
    {
      "title": "col1",
      "field": "col1",
      "dataType": "string"
    },
    {
      "title": "col2",
      "field": "col2",
      "dataType": "number"
    }
  ],
  "data": [
    {
      "col1": "abc",
      "col2": 123
    },
    {
      "col1": "abcd",
      "col2": 1234
    }
  ],
  "options": {
    "paging": true,
    "showTitle": false,
    "search": false,
    "headerStyle": {
      "padding": "4px 8px 4px 8px",
      "textAlign": "left",
      "flexDirection": "row",
      "fontFamily": "\\"Rubik\\", sans-serif",
      "fontWeight": 400,
      "fontSize": "1rem",
      "lineHeight": 1.25,
      "letterSpacing": "0.00938em",
      "borderBottom": "unset"
    }
  },
  "components": {}
}
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
      >
        * File preview for first 100 rows
        <br />
        * Only YYYY-mm-dd HH:MM:SS.sss date format is supported
      </h6>
    </div>
  </div>
</div>
`;

exports[`<ConfigureDatatype /> should match snapshot for invalid file preview 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-10"
  >
    Cannot Preview Data  for filename.error
  </div>
</div>
`;

exports[`<ConfigureDatatype /> should match snapshot for json file preview 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-jsonViewer-5"
    >
      <div
        class="MuiBox-root MuiBox-root-7"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          DataFile:
        </p>
        <div
          class="MuiBox-root MuiBox-root-8"
        >
          <div
            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall"
            >
              filename.json
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-9 makeStyles-jsonContainer-4"
      >
        <div
          class="react-json-view"
          style="font-family: monospace; cursor: default; background-color: rgba(0, 0, 0, 0); position: relative;"
        >
          <div
            class="pretty-json-container object-container"
          >
            <div
              class="object-content"
            >
              <div
                class="object-key-val"
              >
                <span>
                  <span
                    style="display: inline-block; cursor: pointer;"
                  >
                    <div
                      class="icon-container"
                      style="display: inline-block; width: 17px;"
                    >
                      <span
                        class="collapsed-icon"
                      >
                        <svg
                          fill="currentColor"
                          style="vertical-align: top; color: rgb(42, 161, 152); height: 1em; width: 1em; padding-left: 2px;"
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 14l6-6-6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span
                      style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                    >
                      <span
                        class="object-key"
                      >
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                        <span>
                          root
                        </span>
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                      </span>
                      <span
                        style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                      >
                        :
                      </span>
                    </span>
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                    >
                      [
                    </span>
                  </span>
                </span>
                <div
                  class="node-ellipsis"
                  style="display: inline-block; color: rgb(203, 75, 22); font-size: 18px; line-height: 10px; cursor: pointer;"
                >
                  ...
                </div>
                <span
                  class="brace-row"
                >
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 0px;"
                  >
                    ]
                  </span>
                  <div
                    class="object-meta-data"
                    style="display: inline-block; padding: 0px 0px 0px 10px;"
                  >
                    <span
                      class="object-size"
                      style="color: rgba(0, 0, 0, 0.3); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                    >
                      2
                       item
                      s
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
